proxy/nginx/lualib/router.lua

```diff
@@ -30,12 +31,21 @@ local key = prefix .. ":routes:" .. host
 
 -- Try to get target for host
 res, err = red:get(key)
+
+-- Exit if route could not be read
+if err then
+    ngx.log(ngx.ERR, "error reading route: " .. err)
+    ngx.exit(ngx.HTTP_INTERNAL_SERVER_ERROR)
```

Comment:
I'm not really sure we should exit if a random route fails, could you please expand on this one?

Reply:
same here